src/transaction/log_recovery.c

```diff
@@ -3192,725 +3166,716 @@ log_recovery_redo (THREAD_ENTRY * thread_p, const LOG_LSA * start_redolsa, const
 	   * can be changed (e.g., the log record is stored in an archive page and
 	   * in an active page. Later, we try to modify it whenever is possible.
 	   */
+	  {
+	    const auto log_lsa_pageid = log_pgptr_reader.get_pageid ();
+	    const auto log_lsa_offset = log_pgptr_reader.get_offset ();
+	    if (LSA_ISNULL (&lsa) && logpb_is_page_in_archive (log_lsa_pageid))
+	      {
+		lsa.pageid = log_lsa_pageid + 1;
+	      }
 
-	  if (LSA_ISNULL (&lsa) && logpb_is_page_in_archive (log_lsa.pageid))
-	    {
-	      lsa.pageid = log_lsa.pageid + 1;
-	    }
-
-	  if (!LSA_ISNULL (&lsa) && log_lsa.pageid != NULL_PAGEID
-	      && (lsa.pageid < log_lsa.pageid || (lsa.pageid == log_lsa.pageid && lsa.offset <= log_lsa.offset)))
-	    {
-	      /* It seems to be a system error. Maybe a loop in the log */
+	    if (!LSA_ISNULL (&lsa) && log_lsa_pageid != NULL_PAGEID
+		&& (lsa.pageid < log_lsa_pageid || (lsa.pageid == log_lsa_pageid && lsa.offset <= log_lsa_offset)))
+	      {
+		/* It seems to be a system error. Maybe a loop in the log */
 
-	      LSA_SET_NULL (&log_Gl.unique_stats_table.curr_rcv_rec_lsa);
+		LSA_SET_NULL (&log_Gl.unique_stats_table.curr_rcv_rec_lsa);
 
-	      er_log_debug (ARG_FILE_LINE,
-			    "log_recovery_redo: ** System error: It seems to be a loop in the log\n."
-			    " Current log_rec at %lld|%d. Next log_rec at %lld|%d\n", (long long int) log_lsa.pageid,
-			    log_lsa.offset, (long long int) lsa.pageid, lsa.offset);
-	      logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "log_recovery_redo");
-	      LSA_SET_NULL (&lsa);
-	      break;
-	    }
+		er_log_debug (ARG_FILE_LINE,
+			      "log_recovery_redo: ** System error: It seems to be a loop in the log\n."
+			      " Current log_rec at %lld|%d. Next log_rec at %lld|%d\n", (long long int) log_lsa_pageid,
+			      log_lsa_offset, (long long int) lsa.pageid, lsa.offset);
+		logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "log_recovery_redo");
+		LSA_SET_NULL (&lsa);
+		break;
+	      }
+	  }
```

Comment:
for both variables defined at the beginning of the scope; and because it is a logical separate piece of logic which does not have to do with the parts before and after; ideally it should have been a function